### PR TITLE
change iap assert order

### LINKED ISSUE #4467: iap.iap_test: test_main failed
This test failed!

To configure my behavior, see [the Build Cop Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/master/packages/buildcop).

If I'm commenting on this issue too often, add the `buildcop: quiet` label and
I will stop commenting.

---

commit: 2c0d10c2770c48b1944c2821a0475aa5bef51cc1
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/5ee32728-96a9-432b-8807-31068049daef), [Sponge](http://sponge2/5ee32728-96a9-432b-8807-31068049daef)
status: failed
<details><summary>Test output</summary><br><pre>Traceback (most recent call last):
  File "/workspace/iap/iap_test.py", line 61, in test_main
    assert jwt_validation_result[0]
AssertionError: assert None</pre></details>

### PR DESCRIPTION
**Pull request by leahecole**
## Description

Fixes #4467

Note: It's a good idea to open an issue first for discussion.

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: iap/iap_test.py [hunk 1]
```diff
@@ -57,7 +57,9 @@ def test_main(capsys):
 
     jwt_validation_result = validate_jwt.validate_iap_jwt(
         iap_jwt, expected_audience)
+    out, _ = capsys.readouterr()
 
+    assert 'JWT valdiation error' not in out
```
Comment (tmatsuo) [inline iap/iap_test.py:62]: I don't think that stdout contains the error message.

https://github.com/GoogleCloudPlatform/python-docs-samples/blob/dfe31fb7d5d36daf406f2c5a071e37df8dc7865b/iap/validate_jwt.py#L46

It's returning a tuple and `jwt_validation_result[2]` has the error message.

Comment (engelke) [reply iap/iap_test.py:62]: Those two lines are wrong. It looks like they were put in by the merge with master. I'll take them back out.

**Comment by tmatsuo** [comment]
So now this PR doesn't have any changes? I'm not sure if it will solve the original issue.

**Comment by engelke** [comment]
I can't tell from the original issue whether this test really is flaky, or if it was a one-time thing. I can't find other failures of this test. There may be nothing to solve.

**Comment by leahecole** [comment]
Lol alright - I trust you @engelke - I was mostly guessing

**Comment by leahecole** [comment]
@engelke if flaky, is it obvious what the underlying assertion is that we could potentially use with the `backoff` library?

**Comment by engelke** [comment]
There are only two ways I can see for this to fail. First, if the app at gcp-devrel-iap-reflect.appspot.com doesn't return the correct response, or if the URL for the validation certs doesn't return the right certs. I guess the metadata service could also fail to return a valid token when asked. I'd guess the reflection appspot app might be a bit more likely to have a problem, perhaps due to a very slow startup once in a while.

It's hard to see how to fix it without just rerunning the whole test.

**Comment by tmatsuo** [comment]
My guess is that the code throws an exception. More problematically, the test code is throwing away the error, by asserting the first element of the tuple. 

My suggestion is that, first we need to log the error.

**Comment by tmatsuo** [comment]
I don't think we need to "Update branch", feel free to merge. Also feel free to "Update branch" then merge :)

**Pull request merged**